components/organisms/UserSettingsPage/user-settings-page.tsx

```diff
@@ -467,15 +471,24 @@ const UserSettingsPage = ({ user }: UserSettingsPageProps) => {
                 }}
               >
                 <div className="flex flex-col gap-3">
-                  <label className="text-2xl font-normal">Delete Account</label>
+                  <label htmlFor="delete-account-button" className="text-2xl font-normal">
+                    Delete Account
+                  </label>
                   <div className="w-full md:w-96">
                     <Text>
                       Please note that account deletion is irreversible. Proceed only if you are certain about this
                       action.
                     </Text>
                   </div>
                 </div>
-                <Button type="submit" rel="noopener noreferrer" target="_blank" variant="destructive" className="w-max">
+                <Button
+                  id="delete-account-button"
+                  type="submit"
+                  rel="noopener noreferrer"
+                  target="_blank"
+                  variant="destructive"
+                  className="w-max"
+                >
```

Comment:
These changes aren't necessary. The button has button text. The label above shouldn't be a label as button's don't need labels.

```suggestion
                <Button type="submit" rel="noopener noreferrer" target="_blank" variant="destructive" className="w-max">
```